content/language/ruby/configure-ci-cd.md

```diff
@@ -0,0 +1,129 @@
+
+Complete all the previous sections of this guide, starting with [Containerize a Ruby on Rails application](containerize.md). You must have a [GitHub](https://github.com/signup) account and a [Docker](https://hub.docker.com/signup) account to complete this section.
+
+## Overview
+
+In this section, you'll learn how to set up and use GitHub Actions to build and test your Docker image as well as push it to Docker Hub. You will complete the following steps:
+
+1. Create a new repository on GitHub.
+2. Define the GitHub Actions workflow.
+3. Run the workflow.
+
+## Step one: Create the repository
+
+Create a GitHub repository, configure the Docker Hub credentials, and push your source code.
+
+1. [Create a new repository](https://github.com/new) on GitHub.
+
+2. Open the repository **Settings**, and go to **Secrets and variables** >
+   **Actions**.
+
+3. Create a new **Repository variable** named `DOCKER_USERNAME` and your Docker ID as value.
+
+4. Create a new [Personal Access Token (PAT)](../../security/for-developers/access-tokens.md/#create-an-access-token) for Docker Hub. You can name this token `docker-tutorial`. Make sure access permissions include Read and Write.
+
+5. Add the PAT as a **Repository secret** in your GitHub repository, with the name
+   `DOCKERHUB_TOKEN`.
+
+6. In your local repository on your machine, run the following command to change
+   the origin to the repository you just created. Make sure you change
+   `your-username` to your GitHub username and `your-repository` to the name of
+   the repository you created.
+
+   ```console
+   $ git remote set-url origin https://github.com/your-username/your-repository.git
+   ```
+
+7. Run the following commands to stage, commit, and push your local repository to GitHub.
+
+   ```console
+   $ git add -A
+   $ git commit -m "my commit"
+   $ git push -u origin main
+   ```
+
+## Step two: Set up the workflow
+
+Set up your GitHub Actions workflow for building, testing, and pushing the image
+to Docker Hub.
+
+1. Go to your repository on GitHub and then select the **Actions** tab.
+
+2. Select **set up a workflow yourself**.
+
+   This takes you to a page for creating a new GitHub actions workflow file in
+   your repository, under `.github/workflows/main.yml` by default.
+
+3. In the editor window, copy and paste the following YAML configuration.
+
+   ```yaml
+   name: ci
+   
+   on:
+     push:
+       branches:
+         - main
+   
+   jobs:
+     build:
+       runs-on: ubuntu-latest
+       steps:
+         -
+           name: Login to Docker Hub
+           uses: docker/login-action@v3
+           with:
+             username: ${{ vars.DOCKER_USERNAME }}
+             password: ${{ secrets.DOCKERHUB_TOKEN }}
+         -
+           name: Set up Docker Buildx
+           uses: docker/setup-buildx-action@v3
+         -
+           name: Build and push
+           uses: docker/build-push-action@v6
+           with:
+             platforms: linux/amd64,linux/arm64
```

Comment:
Hello @craig-osterhout,

I've encountered the same error, which is a bit puzzling. I'm working on resolving it, but in the meantime, could we consider a plan B? It would be great if we could manually create the image and push it to Docker Hub instead of relying on GitHub Actions. This way, we can ensure the process continues smoothly. What do you think?

Reply:
![image](https://github.com/user-attachments/assets/6f812726-b288-485a-a581-98211fb0afe1)
I was able to fix it... I deleted the `linux/arm64` from `platforms` and keep the linux/amd64
![image](https://github.com/user-attachments/assets/a9a3c744-012a-46fa-828a-76afedbfad20)
and it worked.

I updated it in the document. Thanks @craig-osterhout and let me know what you think.